// redux-polyglot/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.PropType = undefined;

var _middleware = require('./middleware');

Object.keys(_middleware).forEach(function (key) {
    if (key === "default" || key === "__esModule") return;
    Object.defineProperty(exports, key, {
        enumerable: true,
        get: function get() {
            return _middleware[key];
        }
    });
});

var _actions = require('./actions');

Object.keys(_actions).forEach(function (key) {
    if (key === "default" || key === "__esModule") return;
    Object.defineProperty(exports, key, {
        enumerable: true,
        get: function get() {
            return _actions[key];
        }
    });
});

var _constants = require('./constants');

Object.keys(_constants).forEach(function (key) {
    if (key === "default" || key === "__esModule") return;
    Object.defineProperty(exports, key, {
        enumerable: true,
        get: function get() {
            return _constants[key];
        }
    });
});

var _reducer = require('./reducer');

Object.keys(_reducer).forEach(function (key) {
    if (key === "default" || key === "__esModule") return;
    Object.defineProperty(exports, key, {
        enumerable: true,
        get: function get() {
            return _reducer[key];
        }
    });
});

var _selectors = require('./selectors');

Object.keys(_selectors).forEach(function (key) {
    if (key === "default" || key === "__esModule") return;
    Object.defineProperty(exports, key, {
        enumerable: true,
        get: function get() {
            return _selectors[key];
        }
    });
});

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var PropType = _propTypes2.default.shape({
    t: _propTypes2.default.func.isRequired,
    tc: _propTypes2.default.func.isRequired,
    tt: _propTypes2.default.func.isRequired,
    tu: _propTypes2.default.func.isRequired,
    tm: _propTypes2.default.func.isRequired
});

exports.PropType = PropType;
